File: 800/2104A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        long long a, b, c;
        cin >> a >> b >> c;

        long long sum = a + b + c;

        if (sum % 3 != 0)
            cout << "NO\n";
        else 
        {
            long long avg = sum / 3;

            if (a > avg || b > avg)
                cout << "NO\n";
            else
                cout << "YES\n";
        }
    }

    return 0;
}

File: 800/2104A/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            int a = sc.nextInt();
            int b = sc.nextInt();
            int c = sc.nextInt();
            
            if((a+b+c)%3 != 0)
                System.out.println("NO");
            else
            {
                int avg = (a+b+c)/3;
                
                if(a > avg || b > avg)
                    System.out.println("NO");
                else
                    System.out.println("YES");
            }
        }
    }
}